src/renderer/index.js

```diff
@@ -974,34 +922,31 @@ var Renderer = Base.extend('Renderer', {
             config.halign = 'left';
         } else if (isDataRow) {
             isSelected = isCellSelected || isRowSelected || isColumnSelected;
+        } else if (isFilterRow) {
+            isSelected = false;
+        } else if (isColumnSelected) {
+            isSelected = true;
         } else {
-            format = cellEvent.subgrid.format; // subgrid format can override column format
-            if (isFilterRow) {
-                isSelected = false;
-            } else if (isColumnSelected) {
-                isSelected = true;
-            } else {
-                isSelected = selectionModel.isCellSelectedInColumn(x); // header or summary or other non-meta
-            }
+            isSelected = selectionModel.isCellSelectedInColumn(x); // header or summary or other non-meta
         }
 
         // Set cell contents:
         // * For all cells: set `config.value` (writable property)
         // * For cells outside of row handle column: also set `config.dataRow` for use by valOrFunc
-        if (!isHandleColumn) {
-            //Including hierarchyColumn
-            config.dataRow = cellEvent.dataRow;
-            config.value = cellEvent.value;
-        } else {
+        // * For non-data row tree column cells, do nothing (these cells render blank so value is undefined)
+        if (isHandleColumn) {
             if (isDataRow) {
                 // row handle for a data row
                 if (config.rowHeaderNumbers) {
-                    config.value = r + 1; // row number is 1-based
+                    value = r + 1; // row number is 1-based
                 }
             } else if (isHeaderRow) {
                 // row handle for header row: gets "master" checkbox
                 config.allRowsSelected = selectionModel.areAllRowsSelected();
             }
+        } else if (isDataRow || !cellEvent.isTreeColumn) {
```

Comment:
Per @nmichaud: This voids rendering tree column header which is no good. In perspective we were blanking it in `getCell` but that doesn't mean we will never want to render it. Please restore.